### PR TITLE
Cleanup USE_TENSORRT macro

### PR DESCRIPTION
**Pull request by chilo-ms**
**Description**:  This PR is the beginning step for achieving combined Nvidia GPU package.
We started to remove USE_TENSORRT macro first , so later we can integrate TRT EP into existing GPU package without breaking existing CI building/testing behaviors as before. 

Note: It's been tested with TRT8.

### CODE REVIEW

File: onnxruntime/test/providers/cpu/model_tests.cc [hunk 1]
```diff
@@ -66,6 +66,8 @@ TEST_P(ModelTest, Run) {
   if (model_info->GetONNXOpSetVersion() != 8 && provider_name == "tensorrt") {
     // TensorRT can run most of the model tests, but only part of
     // them is enabled here to save CI build time.
+    // Besides saving CI build time, TRT isn’t able to support full ONNX ops spec and therefore some testcases will fail.
```
Comment (snnn) [inline onnxruntime/test/providers/cpu/model_tests.cc:69]: How about we always put the focus on the latest op set? I think TRT users don't quite care about opset 8. @jywu-msft , what do you think?

File: onnxruntime/test/providers/cpu/model_tests.cc [hunk 2]
```diff
@@ -524,7 +548,31 @@ TEST_P(ModelTest, Run) {
       } else if (provider_name == "nuphar") {
         ASSERT_STATUS_OK(session_object.RegisterExecutionProvider(DefaultNupharExecutionProvider()));
       } else if (provider_name == "tensorrt") {
-        ASSERT_STATUS_OK(session_object.RegisterExecutionProvider(DefaultTensorrtExecutionProvider()));
+        if (test_case_name.find(ORT_TSTR("FLOAT16")) != std::string::npos) {
```
Comment (snnn) [inline onnxruntime/test/providers/cpu/model_tests.cc:551]: I heard M60 doesn't have good support for FLOAT16. Would it be the reason why they are failing? Do you want to change the GPU to a different type?

File: onnxruntime/test/util/include/default_providers.h [hunk 3]
```diff
@@ -33,6 +33,7 @@ std::unique_ptr<IExecutionProvider> DefaultCudaExecutionProvider();
 std::unique_ptr<IExecutionProvider> DefaultDnnlExecutionProvider(bool enable_arena = true);
 std::unique_ptr<IExecutionProvider> DefaultNupharExecutionProvider(bool allow_unaligned_buffers = true);
 std::unique_ptr<IExecutionProvider> DefaultTensorrtExecutionProvider();
```
Comment (snnn) [inline onnxruntime/test/util/include/default_providers.h:35]: Could you please add a comment to "DefaultTensorrtExecutionProvider" and mark it as deprecated? I guess one is enough.

File: onnxruntime/test/providers/cpu/model_tests.cc (hunk 1 above)
Comment (jywu-msft) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: seems like a good suggestion. opset8 was chosen a long time ago (i think it was the latest at the time)

File: onnxruntime/test/providers/cpu/model_tests.cc (hunk 2 above)
Comment (jywu-msft) [reply onnxruntime/test/providers/cpu/model_tests.cc:551]: I think we're using m60 for TRT CI pipeline and T4 for TRT packaging pipeline? 
how do we make the tests pass regardless of which GPU we use? I think there are some api's which can tell us if fp16 is supported by the hardware. we can refine this in future PR?

Comment (snnn) [reply onnxruntime/test/providers/cpu/model_tests.cc:551]: Sure.

Comment (chilo-ms) [reply onnxruntime/test/providers/cpu/model_tests.cc:551]: Yes, TensorRT has api to do this. If we want to detect whether fp16 is available for underline GPU device during test time, TRT EP needs to expose this api as well.

File: onnxruntime/test/providers/cpu/model_tests.cc (hunk 1 above)
Comment (chilo-ms) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: From official website https://github.com/onnx/onnx/blob/master/docs/Versioning.md
I saw the latest op set is 15, but the latest op set in our test models is 13.

So I will test only on opset 13 for TRT EP. One thing to notice is that there are around 68 testcases failed on testing opset 13.

Comment (jywu-msft) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: TensorRT 8.0 is supposed to have some opset 13 support, but 68 test case failures is quite a lot.
just curious how some of the other opsets look, like 11 for example.

Comment (chilo-ms) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: After we have disabled some testcases, opset 13 still has 68 failed testcases, opset 12 has 74 failed testcases and opset 11 has 12 failed testcases. 

I'm testing on CI VM with T4.

File: onnxruntime/test/util/include/default_providers.h (hunk 3 above)
Comment (chilo-ms) [reply onnxruntime/test/util/include/default_providers.h:35]: I was wondering why can't we keep it? Right now several testcases are using this function. Also, if we later deprecate this function, we will need to write provider option initialization code which is lengthy each time under the situation we need default configuration.

**Comment by chilo-ms** [comment]
The fails of orttraining pipeline seem are not cause by my changes. 
I got the same error message on previous build run and re-run the build solves the problem. 
But this time I can't solve it by re-running it.

I found other [PR ](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=459007&view=results)has this pipeline error as well.

**Comment by jywu-msft** [comment]
> The orttraining pipelines' failures seem not cause by my changes.
> I got the same error message on previous pipeline run and re-running the build solves the problem.
> But this time I can't solve it by re-running it.
> 
> I found other [pipeline run](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=459007&view=results) has this error as well.

yes, i've seen this pretty frequently. 
that may be why the pipeline is not currently required for merge.

File: onnxruntime/test/providers/cpu/model_tests.cc (hunk 1 above)
Comment (snnn) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: I just added 14/15 there.

Comment (chilo-ms) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: How can I update model folder to include opset14&15 models on the VM which uses old CI image?

Comment (chilo-ms) [reply onnxruntime/test/providers/cpu/model_tests.cc:69]: Update "broken_tests" for the newly added opsets.

**Review by snnn — APPROVED** [review]
Thank you!

**Review by jywu-msft — APPROVED** [review]

**Comment by snnn** [comment]
Should we merge this PR?

**Comment by chilo-ms** [comment]
/azp run orttraining-linux-ci-pipeline,orttraining-mac-ci-pipeline,orttraining-linux-gpu-ci-pipeline,centos7_cpu,Linux CPU Minimal Build E2E CI Pipeline,Linux Nuphar CI Pipeline,MacOS NoContribops CI Pipeline,Linux OpenVINO CI Pipeline,orttraining-distributed, orttraining-amd-gpu-ci-pipeline, Linux CPU CI Pipeline,Linux CPU x64 NoContribops CI Pipeline,Linux GPU CI Pipeline,Linux GPU TensorRT CI Pipeline,MacOS CI Pipeline,MacOS NoContribops CI Pipeline,Windows CPU CI Pipeline,Windows GPU CI Pipeline,Windows GPU TensorRT CI Pipeline

**Pull request merged**